### ISSUE TITLE
Caffe fully connected (InnerProduct) layers cause ParseException

### ISSUE THREAD

**Issue by the-promised-LAN** [ARM-software/armnn#449]
Hello!

**TLDR:** 
armnnCaffeParser throws _armnn::ParseException: Failed to parse layer [fc9]_ while parsing a Fully Connected layer of a Caffe model

**Details:** 
Following one of the [ArmNN tutorials](https://community.arm.com/developer/tools-software/graphics/b/blog/posts/creating-ar-ml-demo-on-mobile-devices), I'm trying to get a trained YOLOv1 Caffe model working with ArmNN. The model is parsed fine up until it hits a Dense/InnerProduct layer and throws an error. Same error is raised with different YOLOv1 models (converted from Darknet to Caffe) from the same [source](https://github.com/tsingjinyun/caffe-yolov2) as the tutorial. 

Tracked it down to [_ParseFromArray()_ from Google's _protobuf_ library returning false inside _RecordByRecordCaffeParser.cpp_](https://github.com/ARM-software/armnn/blob/ba163f93c8a0e858c9fb1ea85e4ac34c966ef38a/src/armnnCaffeParser/RecordByRecordCaffeParser.cpp#L546), but not sure why.

Is it possible for there to be a regression issue with Dense layer support with Caffe (granted the tutorial was from 2018), or more of a darknet-to-caffe conversion failure?

Thanks! 
(If this question doesn't fit the guidelines, feel free to remove)

**Comment by FinnWilliamsArm** [comment]
Hi the-prmoised_LAN

What Arm NN version are you using?

**Comment by the-promised-LAN** [comment]
Thanks for the reply, FinnWilliamsArm!

I'm using v.20.05 for both ArmNN and Arm Compute Library. The full stack is as follows:

Android NDK: _r17b + Android 26 API libc++ standalone toolchain_
Boost: _v.1.64_
Protobuf: _v.2.6.1 (x86 for arm version)_
Caffe: _v1.0_

**Comment by FinnWilliamsArm** [comment]
Hi again @the-promised-LAN 

I tried your model, I believe your Protobuf might be a bit out of date. Could you update it to v3.5.0 and give it another go?

**Comment by the-promised-LAN** [comment]
Hi @FinnWilliamsArm ,

Many thanks for the help, updating to the latest stack fixed it! Hit me soon after I posted my stack that protobuf might be the issue. 

There is now a separate issue with armnn:IRuntime->EnqueueWorkload that causes app crash, but that will be a separate ticket. Thanks again!
